WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java

```diff
@@ -263,46 +263,35 @@ public void onClick(View view) {
             rootView.findViewById(R.id.postFormat).setVisibility(View.GONE);
         } else {
             // Default values
-            mPostFormatTitles = getResources().getStringArray(R.array.post_formats_array);
-            mPostFormats = new String[]{"aside", "audio", "chat", "gallery", "image", "link", "quote", "standard",
-                    "status", "video"};
+            mPostFormatKeys = new ArrayList<>(Arrays.asList(getResources().getStringArray(R.array.post_format_keys)));
+            mPostFormatNames = new ArrayList<>(
+                    Arrays.asList(getResources().getStringArray(R.array.post_format_display_names)));
             // If we have specific values for this site, use them
             List<PostFormatModel> postFormatModels = mSiteStore.getPostFormats(mSite);
-            if (postFormatModels.size() > 0) {
-                int maxElements = postFormatModels.size();
-                if (SiteUtils.isAccessibleViaWPComAPI(mSite)) {
-                    maxElements += 1;
-                }
-                mPostFormats = new String[maxElements];
-                mPostFormatTitles = new String[maxElements];
-                int i = 0;
-                // Inject the "Standard" post format here since .com response doesn't include it
-                if (SiteUtils.isAccessibleViaWPComAPI(mSite)) {
-                    mPostFormats[i] = "standard";
-                    mPostFormatTitles[i] = getResources().getString(R.string.post_format_standard);
-                    i += 1;
-                }
-                for (PostFormatModel postFormatModel : postFormatModels) {
-                    mPostFormats[i] = postFormatModel.getSlug();
-                    mPostFormatTitles[i] = postFormatModel.getDisplayName();
-                    i += 1;
-                }
+            for (PostFormatModel postFormatModel : postFormatModels) {
+                mPostFormatKeys.add(postFormatModel.getSlug());
```

Comment:
Here we're not replacing existing values, so we end up with many duplicate post formats.